test/cpp_extensions/cudnn_extension.cpp

```diff
@@ -0,0 +1,73 @@
+#include <torch/torch.h>
+#include <cuda.h>
+#include <torch/torch.h>
+#include <ATen/ATen.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/cudnn/Descriptors.h>
+#include <ATen/cudnn/Exceptions.h>
+#include <ATen/cudnn/Handles.h>
+#include <torch/csrc/cuda/cuda_check.h>
+#include <torch/torch.h>
+#include <iostream>
+
+using namespace at;
+using namespace std;
+using namespace at::native;
+const char *cudnn_relu_name = "cudnn_relu";
+
+// Check arguments to cudnn_relu
+int cudnn_relu_check(const Tensor &inputs, const Tensor &outputs) {
```

Comment:
You never use the return value here. Also, you should be using `AT_CHECK`, see here https://github.com/pytorch/pytorch/blob/master/test/cpp_extensions/cuda_extension.cpp#L9, as it gives nice stack traces

Reply:
Using AT_CHECK for what exactly? You mean if I was to use the return value? If the plan is just to throw errors instead of returning error codes I'll just make everything void.

TensorUtils is a really nice and consistent way of checking parameters and formatting messages. I added a test that the runtime error has a nice message.

Reply:
Good point, I guess `checkContiguous` et al. already do proper checking and error handling. Ignore this comment then, my misunderstanding

Reply:
You end up with nicely formatted errors and it is a central place to make those formatting decisions. However, not a lot of documentation or examples.